python/hail/docs/hailpedia/matrixtable.rst

```diff
@@ -0,0 +1,404 @@
+
+Each method can take either strings referring to top-level fields, an attribute
+reference (useful for accessing nested fields), as well as keyword arguments
+``KEY=VALUE`` to compute new fields. The Python unpack operator ``**`` can be
+used to specify that all fields of a Struct should become top level fields.
+However, be aware that all top-level field names must be unique. In this
+example, `**mt['info']` would fail because `DP` already exists as an entry
+field.
+
+The example below will keep the row keys `locus` and `alleles` as well as add
+two new fields: `AC` is making the subfield `AC` into a top level field and
+`n_filters` is a new computed field.
+
+    >>> mt_new = mt.select_rows(AC = mt.info.AC,
+    ...                         n_filters = hl.len(mt['filters']))
+
+The order of the fields entered as arguments will be maintained in the new
+matrix table.
+
+**Drop**
+
+The complement of `select` methods, :meth:`.MatrixTable.drop` can remove any top
+level field. An example of removing the `GQ` entry field is:
+
+    >>> mt_new = mt.drop('GQ')
+
+**Explode**
+
+Explode operations can is used to unpack a row or column field that is of type array or
+set.
+
+- :meth:`.MatrixTable.explode_rows`
+- :meth:`.MatrixTable.explode_cols`
+
+One use case of explode is to duplicate rows:
+
+    >>> mt_new = mt.annotate_rows(replicate_num = [1, 2])
+    >>> mt_new = mt_new.explode_rows(mt_new['replicate_num'])
+    >>> mt.count_rows()
+    346
+    >>> mt_new.count_rows()
+    692
+
+    >>> mt_new.replicate_num.show()
+    +---------------+------------+---------------+
+    | locus         | alleles    | replicate_num |
+    +---------------+------------+---------------+
+    | locus<GRCh37> | array<str> |         int32 |
+    +---------------+------------+---------------+
+    | 20:10019093   | ["A","G"]  |             1 |
+    | 20:10019093   | ["A","G"]  |             2 |
+    | 20:10026348   | ["A","G"]  |             1 |
+    | 20:10026348   | ["A","G"]  |             2 |
+    | 20:10026357   | ["T","C"]  |             1 |
+    | 20:10026357   | ["T","C"]  |             2 |
+    | 20:10030188   | ["T","A"]  |             1 |
+    | 20:10030188   | ["T","A"]  |             2 |
+    | 20:10030452   | ["G","A"]  |             1 |
+    | 20:10030452   | ["G","A"]  |             2 |
+    +---------------+------------+---------------+
+
+Aggregation
+===========
+
+:class:`.MatrixTable` has three methods to compute aggregate statistics.
+
+- :meth:`.MatrixTable.aggregate_rows`
+- :meth:`.MatrixTable.aggregate_cols`
+- :meth:`.MatrixTable.aggregate_entries`
+
+These methods take an aggregated expression and evaluate it, returning
+a Python value.
+
+An example of querying entries is to compute the global mean of field `GQ`:
+
+    >>> mt.aggregate_entries(hl.agg.mean(mt.GQ))
+    67.73196915777027
+
+It is possible to compute multiple values simultaneously (and encouraged,
+because grouping two computations together will run twice as fast!) by
```

Comment:
running twice as fast is not strictly true, maybe we can say something like "because grouping two computations together is far more efficient, traversing the dataset only once rather than twice.".